src/lib/homestore.cpp

```diff
@@ -147,37 +151,93 @@ bool HomeStore::start(const hs_input_params& input, hs_before_services_starting_
 }
 
 void HomeStore::format_and_start(std::map< uint32_t, hs_format_params >&& format_opts) {
-    auto total_pct_sum = 0.0f;
-    for (const auto& [svc_type, fparams] : format_opts) {
-        total_pct_sum += fparams.size_pct;
+
+    auto get_optimal_dev_type = [this](const uint32_t svc_type) -> HSDevType {
+        const auto& inp_params = HomeStoreStaticConfig::instance().input;
+        if (svc_type & HS_SERVICE::META || svc_type & HS_SERVICE::LOG || svc_type & HS_SERVICE::INDEX) {
+            if (inp_params.has_fast_dev()) {
+                return HSDevType::Fast;
+            } else {
+                return HSDevType::Data;
+            }
+        } else {
+            return HSDevType::Data;
+        }
+    };
+    std::map< HSDevType, float > total_pct_by_type = {{HSDevType::Fast, 0.0f}, {HSDevType::Data, 0.0f}};
+    // applying optimal placement for auto
+    for (auto& [svc_type, fparams] : format_opts) {
+        if ((svc_type & HS_SERVICE::META) && has_meta_service()) {
+            if (fparams.dev_type == HSDevType::Auto) { fparams.dev_type = get_optimal_dev_type(svc_type); }
```

Comment:
To simplify the logic a bit, I am thinking can we relax this Auto check, and call get_optimal_dev_type directly for every services, the logic is consumer can specify FAST, but if fast is not available, we fall back to DATA. 

This is the logic we apply for 1.3 HomeStore also. Does it make sense?